### DISCUSSION TITLE
How to extract the h264 samples in the media-receiver example

### DISCUSSION THREAD

**Discussion by AlexSZYDLO** [paullouisageneau/libdatachannel#815] [Q&A]
Hello all,

The media-receiver example is working fine and acquires data from the webcam via the browser, that I can visualize with the provided GStreamer pipeline

However, I need to extract the h264 samples in the code. In the example, it looks like the data is RTP packets, which I have no idea how to convert into h264...

I'm not sure if I'm missing something or if this is just not implemented in this lib. I believe the missing part is what gstreamer's "rtph264depay" filter is doing but the code looks pretty complex...

Any thoughts or help would be much appreciated... At least to confirm that I'm looking in the right direction :)

Thanks!

**Comment by feiwei9696** [comment]
I am wondering for the same question, it seems only implemented packetizations for send direction, but not implemented depacketizations yet for recv direction. 
Is any plan for the depacketizations? @paullouisageneau

**Reply by AlexSZYDLO** [reply]
FYI, I took the code from here https://github.com/ireader/media-server/blob/master/librtp/payload/rtp-h264-unpack.c and adapted it to extract the h264 data

**Reply by feiwei9696** [reply]
thanks a lot!
how about the rtcp feedback for nack, receive report?

**Reply by AlexSZYDLO** [reply]
I don't know about that...

**Reply by feiwei9696** [reply]
how do you deal with the missing packets when receive rtp packets?

**Reply by AlexSZYDLO** [reply]
I don't, I just send everything to ffmpeg for decoding, I believe it handles it...

**Reply by paullouisageneau** [reply]
There is currently no support for depacketization (see https://github.com/paullouisageneau/libdatachannel/issues/676) but I'd like to add support in the future. To do it properly, it requires a jitter buffer to reorder and reassemble frames, plus support for detecting loss and sending NACK and PLI.

**Reply by feiwei9696** [reply]
looking forward to :)
